test/src/test/java/hudson/model/UpdateCenterErrorsTest.java

```diff
@@ -0,0 +1,185 @@
+     */
+    @Test
+    public void updateSiteReturn502Test() throws Exception {
+        checkUpdateSite(Jenkins.getInstance().getRootUrl() + "updateSite502/getJson", "Server returned HTTP response code: 502 for URL", false);
+    }
+
+    /**
+     * Check if the page contains the right message after checking an update site with an url that returns a wrong json.
+     * @throws Exception If there are errors managing the web elements.
+     */
+    @Test
+    public void updateSiteWrongJsonTest() throws Exception {
+        checkUpdateSite(Jenkins.getInstance().getRootUrl() + "updateSiteWrongJson/getJson", "Unquotted string 'wrongjson'", false);
+    }
+
+    /**
+     * Check if the page contains the right message after checking an update site that returns a well defined json.
+     * @throws Exception If there are errors managing the web elements.
+     */
+    @Test
+    public void updateSiteRightJsonTest() throws Exception {
+        //Avoid CertPathValidatorException: Algorithm constraints check failed on signature algorithm: MD5withRSA
+        DownloadService.signatureCheck = false;
+        // Have to end in update-center.json or it fails. See UpdateSite#getMetadataUrlForDownloadable
+        checkUpdateSite(Jenkins.getInstance().getRootUrl() + "updateSiteRightJson/update-center.json", "", true );
+    }
+
+    private HtmlButton getCheckNow(HtmlPage page){
+        DomNodeList<HtmlElement> elements = page.getElementById("bottom-sticker").getElementsByTagName("button");
+        assertEquals(1, elements.size());
+        return (HtmlButton) elements.get(0);
+    }
+
+    /**
+     * Check the update site.
+     * @param urlUpdateSite If null, use the default update site, otherwise, use this update site.
+     * @param message The message that should exist or not in the page after checking the update site.
+     * @param isSuccess If true, test that PluginManager.CheckUpdateSesrverError + message doesn't exist in the page.
+     *                  If false, test that PluginManager.CheckUpdateSesrverError + message exists in the page.
+     * @throws IOException If an exception is thrown using the UI.
+     * @throws SAXException If an exception is thrown using the UI.
+     */
+    private void checkUpdateSite(String urlUpdateSite, String message, boolean isSuccess) throws IOException, SAXException {
+        if(urlUpdateSite != null) {
+            Jenkins.getActiveInstance().getUpdateCenter().getSites().clear();
+            UpdateSite us = new UpdateSite("CustomUpdateSite", urlUpdateSite);
+            Jenkins.getActiveInstance().getUpdateCenter().getSites().add(us);
+        }
+
+        JenkinsRule.WebClient wc = j.createWebClient();
+        wc.getOptions().setThrowExceptionOnFailingStatusCode(false);
+        HtmlPage p = wc.goTo("pluginManager");
+        Page pageAfterClick = HtmlElementUtil.click(getCheckNow(p));
+        String page = pageAfterClick.getWebResponse().getContentAsString();
+
+        if(isSuccess)
```

Comment:
nit: add `{...}` (and spaces ^^)
or put `assertEquals(isSuccess, page.contais...)` to simplify ?

Reply:
+1

Reply:
+1

Reply:
Just a meta-comment: can we maybe use the embedded "Pick your reaction" to add _+1_ and the likes, so that we 1) get less notifications and 2) the PR grows a _bit_ less in height? Thanks :) :hugs: